### DISCUSSION TITLE
Rendering Tilemaps on Screen instead of World

### DISCUSSION THREAD

**Discussion by JumpLink** [excaliburjs/Excalibur#2761] [Q&A] [answered]
Hi @eonarheim and community,

I have a question regarding rendering tilemaps. Specifically, is it possible to render them directly on the screen coordinate plane? I'm trying to use tiles to draw the background of a menu and have been using the `ex.TileMap` class for this. I've managed to display it on the `Screen` instead of the `World`, which is what I wanted. However, depending on the player's world position, some tiles get hidden (so the world coordinates are probably still used here to determine whether a tile is outside the visible area).

This is how I am currently setting the coordinate plane for the tilemap:

```ts
    // Set the tilemap coordinate plane to the screen
    let trans = tilemap.get(TransformComponent);
    if (trans) {
      trans.coordPlane = CoordPlane.Screen;
    }
```

Is the TileMap class possibly not suitable for this use case? Should I consider writing a custom `ScreenTileMap` class, see if I can fix it in Excalibur or is there a straightforward way to prevent individual tiles from being hidden?

Any advice or insights on this would be much appreciated.

Thank you!

**Comment by JumpLink** [comment, marked as answer]
I have now created an issue for that: https://github.com/excaliburjs/Excalibur/issues/2763

**Reply by eonarheim** [reply]
@JumpLink this is a good question, I think this was a miss in our design. There is some custom offscreen handling and we didn't consider the screen case.

I agree that you should be able to use a TileMap in either coordinate space
